### ISSUE TITLE
Lwt_process.pread raises Lwt_io.Channel_closed("input") when timed out

### ISSUE THREAD

**Issue by NathanReb** [ocsigen/lwt#153]
When using the timeout option of Lwt_process.pread it's possible to get a Lwt.ioChannel_closed("input") exception instead of the expected partial output.

e.g.
`Lwt_process.pread ~timeout:60.0 cmd >>= fun str ->
  parse str |> return |> Lwt_main.run`
raises

```
Fatal error: exception Lwt_io.Channel_closed("input")
Raised at file "src/core/lwt.ml", line 788, characters 22-23
Called from file "src/unix/lwt_main.ml", line 34, characters 8-18
```

cc @braibant

**Comment by aantron** [comment]
Hi, I see this, but I'm not sure what the API *should* do.

It seems like returning partial output can be useful, but it also seems important to know that the process timed out. It does seem `Lwt_io.Channel_closed` is the wrong exception to use for that, though. This issue also definitely involves some documentation that is missing from `Lwt_process`.

If you are still interested, can you say why `Lwt_process.pread` should provide partial output?

Depending on what we think is the right API, this may involve a breaking change. `Lwt_process` does seem to have a lot of corner cases that aren't well-addressed.

**Comment by aantron** [comment]
Addressed in #466.
